/*
* Copyright (c) 2022 Intel Corporation
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*      http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.intel.examples.objectDetection";
option java_outer_classname = "objectDetectionProto";
option objc_class_prefix = "HLW";

package objectDetection;

// The Detection service definition.
service Detection {
  //
  rpc getPredictions (RequestBytes) returns (PredictionsList) {}
}


message RequestBytes {
  bytes data = 1;
  int32 length = 2;
}

// The response message with list of Predictions.
message PredictionsList {
  repeated Prediction predictions = 1;
}

message Prediction {
    int32 x_min = 1;
    int32 y_min = 2;
    int32 x_max = 3;
    int32 y_max = 4;
    float confidence = 5;
    int32 class_id = 6;
}